File: MyClinic/src/app/features/models/patient.ts
export class Patient {
  [x: string]: any;
  public static fromJson(json:any):Patient{
    return new Patient(
      json['idPatient'],
      json['codePatient'],
      json['firstname'],
      json['lastname'],
      json['dateNaissance'],
      json['sexe'],
      json['codeAssurance'],
      json['numTel'],
      json['situation'],
      json['adresse'],
      json['ville'],
      json['pays'],
      json['email'],
      json['allergie'],
      json['antecedantPerso'],
      json['anticidentFam'],
      json['maladieCronique'],
      json['createdAt'],
      json['modifiedAt'],
    );
  }
  constructor(
    public idPatient:number,
    public codePatient:string,
    public firstname:string,
    public lastname:string,
    public dateNaissance:Date,
    public sexe:string,
    public codeAssurance:string,
    public numTel:string,
    public situation:string,
    public adresse:string,
    public ville:string,
    public pays:string,
    public email:string,
    public allergie:string,
    public antecedantPerso:string,
    public anticidentFam:string,
    public maladieCronique:string,
    public createdAt:Date,
    public modifiedAt:Date,
  ){}
}

File: gestionPatient/src/main/java/com/gestionpatient/entities/Patient.java
package com.gestionpatient.entities;

import jakarta.persistence.*;
import jakarta.persistence.Id;
import lombok.*;
import org.springframework.data.annotation.*;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;
import java.util.Date;


@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "patient")
@EntityListeners(AuditingEntityListener.class)
public class Patient {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long idPatient;

    @Column(name="codePatient")
    private String codePatient;

    @Column(name = "firstname")
    private String firstname ;

    @Column(name = "lastname")
    private String lastname;

    @Column(name = "dateNaissance")
    private Date dateNaissance;

    @Column(name="sexe",length=1)
    private char sexe;

    @Column(name="codeAssurance")
    private String codeAssurance;

    @Column(name="numTel")
    private String numTel;

    @Column(name="situation")
    private String situation;

    @Column(name="adresse")
    private String adresse;

    @Column(name="ville")
    private String ville;

    @Column(name="pays")
    private String pays;

    @Column(name="email")
    private String email;

    @Column(name="allergie")
    private String allergie;

    @Column(name="antecedantPerso")
    private String antecedantPerso;

    @Column(name="anticidentFam")
    private String anticidentFam;

    @Column(name="maladieCronique")
    private String maladieCronique;

    @CreatedDate
    @Column(name = "createdAt")
    private Date createdAt;

    @LastModifiedDate
    @Column(name = "modifiedAt")
    private Date modifiedAt;

}
